### PR TITLE
Added 802.11w support

### PR DESCRIPTION
**Pull request by singe**
Added 802.11w aka Management Frame Protection support. Needed for proper WPA3.

### CODE REVIEW

File: berate_ap [hunk 1]
```diff
@@ -1860,6 +1873,12 @@ ht_capab=${HT_CAPAB}
 EOF
 fi
 
+if [[ $IEEE80211W -eq 1 ]]; then
```
Comment (cablethief) [inline berate_ap]: What does hostap do by default? you set this to 2 earlier and I dont see it being used anywhere. What is the 2 for?

File: berate_ap [hunk 2]
```diff
@@ -705,6 +706,7 @@ ISOLATE_CLIENTS=0
 SHARE_METHOD=nat
 IEEE80211N=0
 IEEE80211AC=0
+IEEE80211W=0
```
Comment (cablethief) [inline berate_ap:709]: From what Ive seen is this is where defaults are set. and then the variable is just used later. probably leave this blank if 0 isnt the default.

File: berate_ap [hunk 3]
```diff
@@ -1262,10 +1264,17 @@ while :; do
             shift
             WPA_VERSION="$1"
             [[ "$WPA_VERSION" == "2+1" ]] && WPA_VERSION=1+2
+            [[ "$WPA_VERSION" == "3+2" ]] && WPA_VERSION=2+3
             if [[ "$WPA_VERSION" == "3" ]]; then
                 WPA=2
                 KEY_MGMT="SAE"
                 PAIRWISE="CCMP"
+                IEEE80211W="2" #MFP required for WPA3
```
Comment (cablethief) [inline berate_ap:1272]: this is the 2 I mean

**Comment by singe** [comment]
From hostapd.conf (http://w1.fi/cgit/hostap/tree/hostapd/hostapd.conf)

# ieee80211w: Whether management frame protection (MFP) is enabled
# 0 = disabled (default)
# 1 = optional 
# 2 = required
#ieee80211w=0

**Pull request merged**